keras/src/callbacks/orbax_checkpoint.py

```diff
@@ -0,0 +1,702 @@
+import os
+import warnings
+
+import numpy as np
+
+from keras.src import backend
+from keras.src import ops
+from keras.src.api_export import keras_export
+from keras.src.callbacks.monitor_callback import (
+    MonitorCallback,  # For metric monitoring logic
+)
+from keras.src.utils.io_utils import print_msg
+from keras.src.utils.module_utils import LazyModule
+
+ocp = LazyModule(
+    "orbax.checkpoint",
+    pip_name="orbax-checkpoint",
+    import_error_msg=(
+        "OrbaxCheckpoint requires the 'orbax-checkpoint' package. "
+        "Install it with: pip install orbax-checkpoint"
+    ),
+)
+
+# Note: Advanced Orbax functionality is available through the ocp LazyModule
+# Users can access it via: from keras.src.utils.module_utils import LazyModule
+# ocp = LazyModule("orbax.checkpoint"); ocp.CheckpointManager
+
+
+def _get_state_tree(model):
+    """Get the complete model state as a nested tree structure."""
+    state_tree = model.get_state_tree(value_format="numpy_array")
+
+    # Convert numpy scalar types to Python types for Orbax compatibility
+    def convert_scalars(obj):
```

Comment:
Is this for the NumPy backend? Or for the JAX backend?

Reply:
> state_tree = model.get_state_tree(value_format="numpy_array")

Oh, but we can't do "numpy_array", at least not for JAX. Because at that point the sharding is irreversibly lost and this is no longer distributed checkpointing.

> NumPy scalar types and 0-dimensional arrays may not serialize correctly or may cause compatibility issues.

Is this documented? Or is this a bug? Do you understand why?